### ISSUE TITLE
Set DateTime.Kind to Local

### ISSUE THREAD

**Issue by czema** [sebastienros/jint#666]
Our system requires that all dates have Kind set to Local.  When I create a date in Jint it creates DateTime objects with Kind set to Utc.   Is there any possible way to change this behavior?  My understanding is that, in Javascript, `new Date()` is supposed to be local time.

```cs
// "task" is an object which has a DateTime assign_dt property.
var engine = new Engine().SetValue("task", task);
engine.Execute("task.assign_dt = new Date();");
Assert.Equal(task.assign_dt.Kind, DateTimeKind.Local);
```

**Comment by sebastienros** [comment]
Internally the value of a date is stored using an UTC date time, correct. But when you go back to C#, why don't you call ToLocal() to cover it to the local timezone?

**Comment by czema** [comment]
Well, since the script can be authored by anyone I wouldn't know when to call ToLocal.  And there are lots of dates if various objects, I just gave this one as a demonstration.

**Comment by sebastienros** [comment]
But within a script, you should not have to use the .NET internal type, it's an implementation detail.

I could see some ways to change this:
- Change how we store it internally, but that would break when people expect it to be Utc.
- Create you own Date implementation that copies our, and just use Local internally
- Make it an option in Jint, such that the Date code would change its internal behavior, but kind of borderline

**Comment by czema** [comment]
I solved this for myself by forcing all DateTime objects to DateTimeKind.Local when storing them in the database.  This is not ideal, but at least I was able to continue working.

I realize that changing your date implementation would break a bunch of existing applications.  But out of curiosity, why does new Date() create a DateTime in UTC when the JavaScript spec says new Date should create an object that represents date and time in the local time zone?
